pkg/filesystem/path/virtual_root_scope_walker_factory.go

```diff
@@ -22,7 +22,7 @@ type namelessVirtualRootNode struct {
 // that point into the root directory.
 type namedVirtualRootNode struct {
 	nameless namelessVirtualRootNode
-	target   string
+	target   Parser
```

Comment:
By storing `Parser` objects inside these nodes, we end up handing them out repeatedly. I think this is fine right now, because we've implemented `UNIXParser` in such a way that it isn't destructive. That said, should we add some documentation to the `Parser` and `RelativeParser` interfaces to state that they MUST be implemented that way?